py/_io/terminalwriter.py

```diff
@@ -144,12 +144,30 @@ def __init__(self, file=None, stringio=False, encoding=None):
     def fullwidth(self):
         if hasattr(self, '_terminal_width'):
             return self._terminal_width
-        return get_terminal_width()
+        try:
+            width = self._cached_terminal_width
```

Comment:
> the proposed implementation should use getattr instead of hasattr

Where? and why?

> and doesnt take lack of signal into account

It does:

```python
        if hasattr(signal, 'SIGWINCH'):
            signal.signal(signal.SIGWINCH, self._invalidate_cached_terminal_width)
```

Or at least it does, I assume the original try/except for AttributeError was because `signal.SIGWINCH` might not exist.